File: RECURSION/print_1_to_N.cpp
//WAP to print 1 to N using recursion 
#include <iostream>
using namespace std;

void printNumbers(int i, int n) {
    if (i > n) return;
    cout << i << " ";
    printNumbers(i + 1, n);
}

int main() {
    int n;
    cout << "Enter a positive integer: ";
    cin >> n;
    printNumbers(1, n);
    return 0;
}

File: JAVA/RECURSION/print_1_to_N.java
//WAP to print 1 to N using recursion 
import java.util.Scanner;

public class PrintNumbers {
    static void printNumbers(int i, int n) {
        if (i > n) return;
        System.out.print(i + " ");
        printNumbers(i + 1, n);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a positive integer: ");
        int n = sc.nextInt();
        printNumbers(1, n);
        sc.close();
    }
}

